.github/workflows/dynamic-checks.yml

```diff
@@ -0,0 +1,115 @@
+name: Dynamic Analysis Checks
+
+on:
+  workflow_dispatch: # manually
+  schedule: # nightly
+    - cron: "0 2 * * *"
+  pull_request: # on pull requests touching locking-related files
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - '.github/workflows/dynamic-checks.yml'
+      - 'internal/locking/**'
+      - 'ddtrace/tracer/**'
+      - 'scripts/ci_test_*.sh'
+      - '**/go.mod'
+  merge_group:
+  push:
+    branches:
+      - main
+      - release-v*
+    tags-ignore:
+      - 'contrib/**'
+      - 'instrumentation/**'
+
+permissions:
+  contents: read
+
+jobs:
+  test-with-debug:
+    name: Dynamic Analysis with Deadlock Detection
+    uses: ./.github/workflows/unit-integration-tests.yml
+    with:
+      go-version: stable
+      ref: ${{ github.ref }}
+      build_tags: "debug"
+    secrets: inherit
+
+  test-with-deadlock:
+    name: Dynamic Analysis with Debug and Deadlock Detection
+    uses: ./.github/workflows/unit-integration-tests.yml
+    with:
+      go-version: stable
+      ref: ${{ github.ref }}
+      build_tags: "deadlock"
+    secrets: inherit
```

Comment:
This adds another run of `unit-integration-tests` to PRs. I think it'd more efficient to add the build tag to the existing checks.